.github/workflows/live-test.yml

```diff
@@ -1,24 +1,70 @@
 name: Live tests for B2K
 on:
-  push:
-    branches: [ "main" ]
-  pull_request:
-    branches: [ "main" ]
+  repository_dispatch:
+    types: [ok-to-test-command]
+permissions:
+  id-token: write
 jobs:
   liveTest:
+    env:
+      PR_NUMBER: ${{ github.event.client_payload.slash_command.args.named.PR_NUMBER }}
+      HEAD_REPO: ${{ github.event.client_payload.slash_command.args.named.HEAD_REPO }}
+      HEAD_BRANCH: ${{ github.event.client_payload.slash_command.args.named.HEAD_BRANCH }}
     strategy:
       matrix:
-        os: [ubuntu-latest]
+        os: [ubuntu-latest, macos-latest, windows-latest]
     runs-on: ${{ matrix.os }}
     name: live test for b2k
     steps:
     - uses: actions/checkout@v2
+      # checkout PR head repo/branch
+      with:
+         repository: ${{ env.HEAD_REPO }}
+         ref: ${{ env.HEAD_BRANCH }}
+    - name: Login Azure Context
+      if: ${{ matrix.os != 'ubuntu-latest' }}
+      uses: Azure/login@v1
+      with:
+        tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+        subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+        client-id: ${{ secrets.AZURE_CLIENT_ID }}
+    - name: Prepare Cluster name Windows
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: |
+        echo "windows_cluster=b2k-livetest-win-pr-$env:PR_NUMBER" >> $env:GITHUB_ENV
+    # create cluster for Windows
+    - name: Create AKS for Windows
+      if: ${{ matrix.os == 'windows-latest' }}
+      run: |
+        az aks create -g ${{ secrets.AZURE_RESOURCE_GROUP }} -n ${{ env.windows_cluster }} --no-ssh-key
+    - name: Set AKS Context for Windows
+      if: ${{ matrix.os == 'windows-latest' }}
+      uses: azure/aks-set-context@v3
+      with:
+        resource-group: ${{ secrets.AZURE_RESOURCE_GROUP }}
+        cluster-name: ${{ env.windows_cluster }}
+    - name: Prepare Cluster name Mac
+      if: ${{ matrix.os == 'macos-latest' }}
+      run: |
+        echo "mac_cluster=b2k-livetest-mac-pr-$PR_NUMBER" >> $GITHUB_ENV
+    # create cluster for Mac
+    - name: Create AKS mac
+      if: ${{ matrix.os == 'macos-latest' }}
+      run: |
+        az aks create -g ${{ secrets.AZURE_RESOURCE_GROUP }} -n ${{ env.mac_cluster }} --no-ssh-key
+    - name: Set AKS Context for MacOS
+      if: ${{ matrix.os == 'macos-latest' }}
+      uses: azure/aks-set-context@v3
+      with:
+        resource-group: ${{ secrets.AZURE_RESOURCE_GROUP }}
+        cluster-name: ${{ env.mac_cluster }}
```

Comment:
@cxznmhdcxz Can we clean up these AKS clusters after B2K is disconnected or as last step so that we don't use up more resources ? example : ```az aks delete -n cluster-name -g resource-group-name -y --no-wait ``` ?